inc/vcf/error.hpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright 2016 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef VCF_ERROR_HPP
+#define VCF_ERROR_HPP
+
+#include <string>
+#include <stdexcept>
+
+namespace ebi
+{
+  namespace vcf
+  {
+    // parent of vcf errors
+    class Error : public std::runtime_error
+    {
+      public:
+        Error(size_t line) : Error(line, "Error, invalid file.") {}
+
+        Error(size_t line, const std::string &message)
+                : runtime_error(std::string("Line ") + std::to_string(line) + ": " + message),
+                  line(line),
```

Comment:
The are several others throughout the code. I can point them all if you want me to, but I thought it would make checking the review comments in general a nightmare :)